From f2ac9609bf15ab13bead12e6cc574218a629f8b8 Mon Sep 17 00:00:00 2001
From: Vincent ABRIOU <vincent.abriou@st.com>
Date: Tue, 4 Apr 2023 15:10:13 +0200
Subject: [PATCH 3/3] makefile_build: Makefile: remove absolute path from the
 generated files

Use the -ffile-prefix-map option to remove the file absolute path from
the generated objects.

Signed-off-by: Vincent ABRIOU <vincent.abriou@st.com>
---
 makefile_build/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/makefile_build/Makefile b/makefile_build/Makefile
index 2f95342..ffbefac 100644
--- a/makefile_build/Makefile
+++ b/makefile_build/Makefile
@@ -216,17 +216,17 @@ firmware:
 $(LIBEDGETPU_CCOBJS) : $(BUILDDIR)/%.o: %.cc
 	@$(MKDIR)
 	@echo "Compiling $<"
-	@$(CXX) $(LIBEDGETPU_CXXFLAGS) $(LIBEDGETPU_INCLUDES) -c $< -MD -MT $@ -MF $(@:%o=%d) -o $@
+	@$(CXX) $(LIBEDGETPU_CXXFLAGS) $(LIBEDGETPU_INCLUDES) -ffile-prefix-map=$(WORKDIR)= -c $< -MD -MT $@ -MF $(@:%o=%d) -o $@
 
 $(LIBEDGETPU_MAX_CCOBJS) : $(BUILDDIR)/%.o: %.cc
 	@$(MKDIR)
 	@echo "Compiling $<"
-	@$(CXX) $(LIBEDGETPU_CXXFLAGS) $(LIBEDGETPU_INCLUDES) -c $< -MD -MT $@ -MF $(@:%o=%d) -o $@
+	@$(CXX) $(LIBEDGETPU_CXXFLAGS) $(LIBEDGETPU_INCLUDES) -ffile-prefix-map=$(WORKDIR)= -c $< -MD -MT $@ -MF $(@:%o=%d) -o $@
 
 $(LIBEDGETPU_STD_CCOBJS) : $(BUILDDIR)/%-throttled.o: %.cc
 	@$(MKDIR)
 	@echo "Compiling $<"
-	@$(CXX) -DTHROTTLE_EDGE_TPU $(LIBEDGETPU_CXXFLAGS) $(LIBEDGETPU_INCLUDES) -c $< -MD -MT $@ -MF $(@:%o=%d) -o $@
+	@$(CXX) -DTHROTTLE_EDGE_TPU $(LIBEDGETPU_CXXFLAGS) $(LIBEDGETPU_INCLUDES) -ffile-prefix-map=$(WORKDIR)= -c $< -MD -MT $@ -MF $(@:%o=%d) -o $@
 
 libedgetpu: $(LIBEDGETPU_CCOBJS) $(LIBEDGETPU_MAX_CCOBJS)
 	@mkdir -p $(LIBDIR)
-- 
2.25.1

